specifyweb/frontend/js_src/lib/components/QueryComboBox/index.tsx

```diff
@@ -508,12 +508,9 @@ export function QueryComboBox({
                 }
               />
             )}
-            {formatted?.resource === undefined ||
-            (hasViewButton &&
-              hasTablePermission(
-                formatted.resource.specifyTable.name,
-                'create'
-              ))
+            {hasViewButton &&
+            formatted?.resource !== undefined &&
+            hasTablePermission(formatted.resource.specifyTable.name, 'create')
```

Comment:
UX concern here - because you have `formatted?.resource !== undefined`, the button won't be there on blank form, but will suddenly appear when the user selects some resource in the query combo box - we generally try to avoid layout shifts/changes when possible (i.e in this case, the view button appearing suddenly, causing all other buttons to to shift by one suddenly, and potentially causing user to click on a wrong button if they were aiming for a click at that moment)
similarly, when opening an existing resource, `formatted` won't be set right away (since it takes time to fetch the formatted resource), leading to the view buttons appearing with a delay, changing the width of the query combo box, and potentially even changing the width of the entire table column as a consequence

instead, could you please do this?

```suggestion
            hasTablePermission(relatedTable.name, 'create')
```